x/wasm/keeper/relay.go

```diff
@@ -194,7 +200,11 @@ func (k Keeper) OnAckPacket(
 	if execErr != nil {
 		return errorsmod.Wrap(types.ErrExecuteFailed, execErr.Error())
 	}
-	return k.handleIBCBasicContractResponse(ctx, contractAddr, contractInfo.IBCPortID, res)
+	if res.Err != "" {
```

Comment:
should we add a check that `res != nil`?

Reply:
Right now wasmvm only returns `nil` in case of an `err`, but it's probably better to add a check here anyways, just in case.